src/Service/MagicLinks/MagicLinkService.cs

```diff
@@ -0,0 +1,60 @@
+using Passwordless.Common.Services.Mail;
+using Passwordless.Service.MagicLinks.Models;
+using Passwordless.Service.Models;
+
+namespace Passwordless.Service.MagicLinks;
+
+public class MagicLinkService
+{
+    private readonly IFido2Service _fido2Service;
+    private readonly IMailProvider _mailProvider;
+
+    public MagicLinkService(IFido2Service fido2Service, IMailProvider mailProvider)
+    {
+        _fido2Service = fido2Service;
+        _mailProvider = mailProvider;
+    }
+
+    public async Task<MagicLinkResult> SendMagicLink(MagicLinkDTO dto)
+    {
```

Comment:
@Tyrrrz Are we adding the quota checks here?

Reply:
Yes